src/script/view_model/content/PreferencesDeviceDetailsViewModel.ts

```diff
@@ -0,0 +1,132 @@
+import {amplify} from 'amplify';
+import ko from 'knockout';
+
+import {t} from 'Util/LocalizerUtil';
+import {formatTimestamp} from 'Util/TimeUtil';
+
+import {Config} from '../../Config';
+import {MotionDuration} from '../../motion/MotionDuration';
+import {ContentViewModel} from '../ContentViewModel';
+import {MainViewModel} from '../MainViewModel';
+import {ClientRepository} from '../../client/ClientRepository';
+import {ConversationRepository} from '../../conversation/ConversationRepository';
+import {CryptographyRepository} from '../../cryptography/CryptographyRepository';
+import {ClientEntity} from '../../client/ClientEntity';
+import {ActionsViewModel} from '../ActionsViewModel';
+import {User} from '../../entity/User';
+import {getLogger, Logger} from 'Util/Logger';
+
+export class PreferencesDeviceDetailsViewModel {
+  private readonly logger: Logger;
+  private readonly actionsViewModel: ActionsViewModel;
+  private readonly selfUser: ko.Observable<User>;
+  readonly activationDate: ko.Observable<string>;
+  readonly device: ko.Observable<ClientEntity>;
+  readonly fingerprint: ko.ObservableArray<string>;
+  readonly SessionResetState: typeof PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE;
+  readonly sessionResetState: ko.Observable<string>;
+  readonly brandName: string;
+
+  static get SESSION_RESET_STATE() {
+    return {
+      CONFIRMATION: 'confirmation',
+      ONGOING: 'ongoing',
+      RESET: 'reset',
+    };
+  }
+
+  constructor(
+    mainViewModel: MainViewModel,
+    private readonly clientRepository: ClientRepository,
+    private readonly conversationRepository: ConversationRepository,
+    private readonly cryptographyRepository: CryptographyRepository,
+  ) {
+    this.actionsViewModel = mainViewModel.actions;
+    this.selfUser = this.clientRepository.selfUser;
+    this.SessionResetState = PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE;
+    this.logger = getLogger('PreferencesDeviceDetailsViewModel');
+
+    this.activationDate = ko.observable();
+    this.device = ko.observable();
+    this.fingerprint = ko.observableArray([]);
+    this.sessionResetState = ko.observable(PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE.RESET);
+    this.brandName = Config.getConfig().BRAND_NAME;
+
+    this.device.subscribe(clientEntity => {
+      if (clientEntity) {
+        this.sessionResetState(PreferencesDeviceDetailsViewModel.SESSION_RESET_STATE.RESET);
+        this._updateFingerprint();
+        const date = formatTimestamp(clientEntity.time);
+        this.activationDate(t('preferencesDevicesActivatedOn', {date}));
+      }
+    });
+  }
+
+  _updateFingerprint = async (): Promise<void> => {
+    this.fingerprint([]);
+    try {
+      const fingerprint = await this.cryptographyRepository.getRemoteFingerprint(this.selfUser().id, this.device().id);
+      this.fingerprint(fingerprint);
+    } catch (error) {
+      this.logger.warn('Error while trying to update fingerprint', error);
```

Comment:
Before, the error was not caught here. Does it have any implications that you are now catching it?